tests/apps/tests.py

```diff
@@ -49,6 +53,34 @@ def test_ready(self):
         # Non-master app registries are populated in __init__.
         self.assertTrue(Apps().ready)
 
+    def test_populate_reentrance(self):
+        """
+        Running django.setup() during an ongoing setup() results in a deadlock.
+
+        To prevent Django's tests from hanging indefinitely if a deadlock
```

Comment:
The test relied on mocking `threading.RLock`. The original code used a `Lock`, so the mock is ineffective.

Working on this, I remembered why a Thread cannot be used to run the test: it's not possible to kill the test thread when it is stuck. The only (simple) option I can think of is to run the test in another process, using `multiprocessing`. However, when the tests are executed in parallel, the test processes are daemonic and cannot have children, so it is not possible to start a new process to run the test.

Do you have an idea on how to test this patch? Would it be acceptable to merge it without a corresponding test?